lightning/src/chain/keysinterface.rs

```diff
@@ -445,11 +445,20 @@ pub trait KeysInterface {
 	/// This method should return a different value each time it is called, to avoid linking
 	/// on-chain funds across channels as controlled to the same user.
 	fn get_shutdown_scriptpubkey(&self) -> ShutdownScript;
-	/// Get a new set of Sign for per-channel secrets. These MUST be unique even if you
-	/// restarted with some stale data!
+	/// Generates a unique `channel_keys_id` that can be used to obtain a `Signer` through
+	/// [`KeysInterface::derive_channel_signer`]. The `user_channel_id` is provided to allow
+	/// implementations of `KeysInterface` to maintain a mapping between it and the generated
+	/// `channel_keys_id`.
 	///
 	/// This method must return a different value each time it is called.
-	fn get_channel_signer(&self, inbound: bool, channel_value_satoshis: u64) -> Self::Signer;
+	fn generate_channel_keys_id(&self, user_channel_id: u128) -> [u8; 32];
```

Comment:
With the current default `KeysManager` implementation, the new `user_channel_id` enters into the derivation chain for per-channel keys, if this is correct expected behavior for all trait implementations, I think we should document here that `user_channel_id` is a critical piece of data, like a seed and should be backed up accordingly.